File: Examples/CSharp/Articles/ExportVisibleRowsData.cs
using System;
using System.IO;

using Aspose.Cells;
using System.Data;

namespace Aspose.Cells.Examples.CSharp.Articles
{
    public class ExportVisibleRowsData
    {
        public static void Run()
        {
            //Source directory
            string sourceDir = RunExamples.Get_SourceDirectory();

            // Load the source workbook
            Workbook workbook = new Workbook(sourceDir + "sampleExportVisibleRowsData.xlsx");

            // Access the first worksheet
            Worksheet worksheet = workbook.Worksheets[0];

            // Specify export table options
            ExportTableOptions exportOptions = new ExportTableOptions();
            exportOptions.PlotVisibleRows = true;
            exportOptions.ExportColumnName = true;

            // Export the data from worksheet with export options
            DataTable dataTable = worksheet.Cells.ExportDataTable(0, 0, 10, 4, exportOptions);

            Console.WriteLine("ExportVisibleRowsData executed successfully.\r\n");
        }
    }
}

File: Examples/VisualBasic/Articles/ExportVisibleRowsData.vb
Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.Cells
Imports System.Data

Namespace Aspose.Cells.Examples.Articles
    Public Class ExportVisibleRowsData
        Public Shared Sub Main()
            'ExStart:1
            ' The path to the documents directory.
            Dim dataDir As String = Aspose.Cells.Examples.Utils.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)
            Dim filePath As String = dataDir & "aspose-sample.xlsx"

            'Load the source workbook
            Dim workbook As New Workbook(filePath)

            'Access the first worksheet
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            'Specify export table options
            Dim exportOptions As New ExportTableOptions()
            exportOptions.PlotVisibleRows = True
            exportOptions.ExportColumnName = True

            'Export the data from worksheet with export options
            Dim dataTable As DataTable = worksheet.Cells.ExportDataTable(0, 0, 10, 4, exportOptions)
            'ExEnd:1


        End Sub
    End Class
End Namespace